### PR TITLE
Factorial

### LINKED ISSUE #1819: Finding factorial of numbers like 100! (C,C++,JAVA)
Factorial of 100 is too large for a variable to store. I can contribute for this problem. please assign the task to me @singh-shreya6

### PR DESCRIPTION
**Pull request by ritish099**
Fixes #1819

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Languages Used: 
- C
- C++
- JAVA
#### Files Added: ,, Factorial.java

- Factorial.c
- Factorial.cpp
- Factorial.java

Thanks!

### CODE REVIEW

**Comment by ritish099** [comment]
@SakshiShreya kindly review this mam

**Comment by ritish099** [comment]
Issue number #1819

File: Factorial/Factorial.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/* This code is used to find factorial 
+of large numbers*/
+#include<stdio.h>
+int main()
+{
+    int n,a[200000],i,j,k,l,m;
```
Comment (asha15) [inline Factorial/Factorial.c]: keep a space after comma

File: Factorial/Factorial.c [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    printf("ENTER THE NUMBER\n");
+	scanf("%d",&n);
```
Comment (asha15) [inline Factorial/Factorial.c]: keep a space after comma and correct the indentation

File: Factorial/Factorial.c [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        m=1;
+        a[0]=1;
+        for(j=2;j<=n;j++)
```
Comment (asha15) [inline Factorial/Factorial.c]: correct the indentation (keep four spaces) and keep space before and after operators

File: Factorial/Factorial.c [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                l=a[k]/10;
+                a[k]=a[k]%10;
+            }
+            while(l)
+            {
+	        a[k]=l%10;
+                k++;
+                m++;
+                l=l/10;
+            }
```
Comment (asha15) [inline Factorial/Factorial.c]: please keep space before and after operators

Comment (asha15) [reply Factorial/Factorial.c]: Also add some commends regarding the operation

File: Factorial/Factorial.c [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        }
+        printf("Answer is : ");
+        for(i=m-1;i>=0;i--)
+            printf("%d",a[i]);
```
Comment (asha15) [inline Factorial/Factorial.c]: keep space before and after operators, keep space after comma

File: Factorial/Factorial.c [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    return 0;
+}
```
Comment (asha15) [inline Factorial/Factorial.c]: keep a line space after this

File: Factorial/Factorial.cpp [hunk 7]
```diff
@@ -0,0 +1,34 @@
+#include <iostream>
+using namespace std;
+main()
+{
+    int t,n,a[200000],i,j,k,l,m;
```
Comment (asha15) [inline Factorial/Factorial.cpp]: keep space after comma

File: Factorial/Factorial.cpp [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    cout<<"ENTER THE NUMBER\n";
```
Comment (asha15) [inline Factorial/Factorial.cpp]: keep space after the operators

File: Factorial/Factorial.cpp [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+            {
+	            a[k]=l%10;
+                k++;
+                m++;
+                l=l/10;
+            }
+        }
+        printf("Answer is : ");
+        for(i=m-1;i>=0;i--)
+            cout<<a[i];
```
Comment (asha15) [inline Factorial/Factorial.cpp]: correct the indentation

Comment (asha15) [reply Factorial/Factorial.cpp]: keep space before and after operators

Comment (asha15) [reply Factorial/Factorial.cpp]: keep space after comma

File: Factorial/Factorial.cpp [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    return 0;
+}
```
Comment (asha15) [inline Factorial/Factorial.cpp]: keep a line space after this

File: Factorial/Factorial.cpp [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+/*ENTER THE NUMBER
+5
```
Comment (asha15) [inline Factorial/Factorial.cpp]: take this to previous line

File: Factorial/Factorial.c [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+/*ENTER THE NUMBER
+5
```
Comment (asha15) [inline Factorial/Factorial.c]: take this to previous line

File: Factorial/Factorial.java [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+	{
+		    Scanner s = new Scanner(System.in);
+		    int n = s.nextInt();
+		    BigInteger f = new BigInteger("1");
+		    for(int j = 1;j<=n;j++)
+		    {
+		        f = f.multiply(BigInteger.valueOf(j));
+		    }
+		    System.out.println(f);
+	}
```
Comment (asha15) [inline Factorial/Factorial.java]: - correct the indentation. keep four spaces instead to tab
- please keep space before and after operators

File: Factorial/Factorial.java [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+}
```
Comment (asha15) [inline Factorial/Factorial.java]: keep a line space after this

File: Factorial/Factorial.java [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+/*ENTER THE NUMBER
+5
```
Comment (asha15) [inline Factorial/Factorial.java]: take this to the previous line

File: Factorial/Factorial.c (hunk 2 above)
Comment (ritish099) [reply Factorial/Factorial.c]: Done. pls check it out @asha15

File: Factorial/Factorial.java [hunk 16]
```diff
@@ -0,0 +1,21 @@
+/* This code is used to find factorial
+of large numbers*/
```
Comment (asha15) [inline Factorial/Factorial.java]: Take this commend into previous line and keep a line space

File: Factorial/Factorial.java [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+      Scanner s = new Scanner(System.in);
+      int n = s.nextInt();
+      BigInteger f = new BigInteger("1");
+      for (int j = 1; j <= n; j++) 
+      {
+        f = f.multiply(BigInteger.valueOf(j));
+      }
+      System.out.println(f);
+  }
+}/*ENTER THE NUMBER
```
Comment (asha15) [inline Factorial/Factorial.java]: correct the indentation(keep four spaces instead of two)

Comment (asha15) [inline Factorial/Factorial.java]: I/O needs to define at the end

**Comment by ritish099** [comment]
@asha15 look into it

**Comment by asha15** [comment]
Good job :)

**Comment by mahakmandlecha** [comment]
@asha15  ma'am please assign me this task to find factorial of large numbers using data structures in c . I'm a participant of GSSoC.

**Comment by asha15** [comment]
please create a issue for it, then will assign it

**Pull request merged**